pkg/migrations/op_add_column_test.go

```diff
@@ -111,6 +111,82 @@ func TestAddColumn(t *testing.T) {
 				}, res)
 			},
 		},
+		{
+			name: "add column unique",
```

Comment:
I don't think this test adds any new coverage; it passes before and after the changes in the rest of this PR.

These tests and the ones like it in the `op_*_test.go` files are tests for the correctness of the various `pgroll` operations and work with small amounts of data. Issues like https://github.com/xataio/pgroll/issues/644 are only visible when we're working with much larger row counts.

Lock backoff and retry is tested elsewhere (like in `execute_test.go`), but there are no automated tests for this kind of two step constraint validation for other things like ensuring adding `NOT NULL` constraint doesn't lock the table for an extended period of time.

This highlights a gap in our testing, but I don't think we should address it in this PR.

Reply:
Fair point! Removed.